### DISCUSSION TITLE
How to configure the PHP command line on Mac?

### DISCUSSION THREAD

**Discussion by pawell0** [datenstrom/community#864] [Ask a question] [answered]
Has anyone had that kind of experience on Mac (Catalina, Intel) when trying to manage users using terminal - warning: "Datenstrom Yellow requires PHP zip extension! See https://datenstrom.se/yellow/help/troubleshooting"? Test serwer on MAMP, ZIP extension installed and running. I don't get it ...

**Comment by PetersOtto** [comment]
Are you really sure that the php zip extension is installed on your server? I work with Linux and do not know Mamp but it should be possible to check which php modules are installed. Maybe according to this guide? 

https://documentation.mamp.info/en/MAMP-Mac/FAQ/Which-PHP-modules-are-included-in-MAMP/

In my opinion, according to yellow troubleshooting, the php zip module is not installed.

**Reply by pawell0** [reply]
![zip](https://user-images.githubusercontent.com/74900425/235342240-90f7ec8a-a99d-40ce-85dd-18e0a4f4fb62.png)
What do you think ...?

**Reply by PetersOtto** [reply]
Hmm, I think the php zip extension is installed. 
If you restarted the server after installing the php zip extension, it should actually work.

Sorry, I'm out of it :(

**Reply by annaesvensson** [reply]
It looks like the PHP zip extension is available, but only on the web server. 😀

MacOS and MAMP come with their own PHP version(s). So what you get in the web browser could be very different to what you get in the command line. Open a terminal window and type `which php`. If it says "/usr/bin/php" then you are still using the PHP version that comes with MacOS. Type `php -v` to see which version it is and `php -m` to see which modules are available.

**Reply by pawell0** [reply]
This is an important observation! Thank you! I hadn't thought of that (!). This requires changing the configuration in some way.

**Reply by annaesvensson** [reply]
What does the command `ls -l /Applications/MAMP/bin/php` show?

**Reply by pawell0** [reply]
-rw-rw-r--  1 myusername  admin   282226  8 gru 21:57 browscap.ini
-rwxr-xr-x  1 myusername  admin  2207173  8 gru 21:14 composer
drwxrwxr-x  7 myusername  admin      224  9 gru 12:45 php5.4.45
drwxrwxr-x  7 myusername  admin      224  9 gru 12:45 php5.6.40
drwxrwxr-x  7 myusername  admin      224  9 gru 12:45 php7.3.33
drwxrwxr-x  7 myusername  admin      224  9 gru 12:45 php7.4.33
drwxrwxr-x  7 myusername  admin      224  9 gru 12:45 php8.1.13
drwxrwxr-x  7 myusername  admin      224 26 sty 19:16 php8.2.0

**Reply by annaesvensson** [reply]
Which PHP version is currently selected in [MAMP](https://documentation.mamp.info/en/MAMP-Mac/First-Steps/index.html), have a screenshot?

**Reply by pawell0** [reply]
![mamp-php](https://user-images.githubusercontent.com/74900425/235629854-b4e67ccc-bd81-4766-845e-6921b3cdce35.png)

**Reply by pawell0** [reply]
Should I switch it to 8.x.x?

**Reply by schulle4u** [reply]
Please create an alias for the desired php version: 

`alias php='/Applications/MAMP/bin/php7.4.33/bin/php'`

and then you can use Yellow in the command line:

`php yellow.php`

**Note**: The alias won't be permanently saved in your `.profile` configuration, but for this you can check various resources on the web. 🙂

**Reply by pawell0** [reply]
Terminal after trying to create the alias:
`zsh: no such file or directory: /Applications/MAMP/bin/php7.4.33/bin/php `

**Reply by annaesvensson** [reply]
You are 90% there. The path is a wee bit wrong, just use the following:

`alias php='/Applications/MAMP/bin/php/php7.4.33/bin/php'`

**Reply by pawell0** [reply]
Thank you, Anna, I did that before. Anyway - it works. 😉👍

**Comment by schulle4u** [comment]
Only as a side note: You might encounter another szenario with different php versions when running the internal web server using the [serve extension](https://github.com/annaesvensson/yellow-serve). In my development environment I'm running more than one php version in parallel to test against incompatibilities and so on. Each version has its own alias command, e.g. `php8.0`, `php8.1`. However, the serve command still uses the system default php command to start the internal web server, so even if I run `php8.1 yellow.php serve`, yellow still uses the system default php version. You can easily change the corresponding line in `serve.php` to temporarily fix this.

**Reply by schulle4u** [reply]
...or simply update to the latest version which uses the correct php binary during runtime. Awesome! 👍

**Comment by annaesvensson** [comment, marked as answer]
Here's the summary [for a solution with MAMP](https://documentation.mamp.info/index.html).

1. Check which PHP versions are available in MAMP:
   `ls -l /Applications/MAMP/bin/php`
2. Check which PHP version is currently selected in MAMP:
   preferences are accessible in the menu → MAMP → Preferences
3. Create an alias for the desired PHP version, for example:
   `alias php='/Applications/MAMP/bin/php/php7.4.33/bin/php'`
4. Save the alias in your configuration file, this step is optional:
   `nano ~/.profile` 
6. Now you should have the same PHP version in the command line.

**Comment by annaesvensson** [comment]
Tip: You can let us know when your question has been answered, click on Mark as answer.
